### PR TITLE
[INLONG-4674][Manager] Refactor the client by using the Retrofit framework

### LINKED ISSUE #4674: [Improve][Manager] Refactor the client by using the Retrofit framework
### Description

Refactor the client by using the Retrofit framework.

See here: <https://square.github.io/retrofit>

### InLong Component

InLong Manager

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by leosanqing**
### Prepare a Pull Request

- Fixes #4674 

### Motivation

The previous implementation of the client is not particularly good
1. The authentication information of the user's login is spliced ​​in the path each time
2. Param and PathVariable are manually spliced ​​together, and the scalability is not good
3. All the request methods for connecting to the web are all in one class
4. The previous execution of the http request package is not concise enough

### Modifications

1. The processing of authentication information is placed in the interceptor, and then if the authentication information is replaced by a header, it only needs to be modified here
2. The method of client requesting web controller is divided according to business, corresponding to the web aspect, which is convenient for maintenance and search
3. Use declarative annotations to simplify development and maintenance

### Verifying this change

*(Please pick either of the following options)*

- [x] This change is already covered by existing tests, such as:
  - Kafka2HiveExample
  - InnerInlongManagerClientTest

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -130,6 +130,7 @@
         <httpcore.version>4.4.14</httpcore.version>
         <httpclient.version>4.5.13</httpclient.version>
         <okhttp.version>3.14.9</okhttp.version>
+        <retrofit.version>2.9.0</retrofit.version>
```
Comment (healchow) [inline pom.xml:133]: Please add the license and notice (if it has) info for the Retrofit dependency.

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/service/InlongStreamApi.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import retrofit2.http.Body;
+import retrofit2.http.GET;
+import retrofit2.http.POST;
+import retrofit2.http.Path;
+import retrofit2.http.Query;
+
+public interface InlongStreamApi {
+
+    @POST("stream/save")
+    Call<Response<Integer>> createStreamInfo(@Body InlongStreamInfo streamInfo);
```
Comment (healchow) [inline inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/service/InlongStreamApi.java:36]: No need to add `Info` at the end of those methods.

**Review by woofyzhao — APPROVED** [review]

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/inner/InnerInlongManagerClient.java [hunk 3]
```diff
@@ -56,40 +58,57 @@
...
 
+    private final InlongStreamApi inlongStreamApi;
+    private final InlongGroupApi inlongGroupApi;
+    private final StreamSourceApi streamSourceApi;
+    private final StreamTransformApi streamTransformApi;
+    private final StreamSinkApi streamSinkApi;
+    private final WorkflowApi workflowApi;
+
     public InnerInlongManagerClient(ClientConfiguration configuration) {
-        this.host = configuration.getBindHost();
```
Comment (kipshi) [inline inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/inner/InnerInlongManagerClient.java:78]: serviceUrl can be set as host1:port1,host2:port2,   so host, port is needed here

Comment (leosanqing) [reply inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/inner/InnerInlongManagerClient.java:78]: my bad,I'll rollback this

**Review by ciscozhou — APPROVED** [review]

**Review by healchow — APPROVED** [review]
LGTM

**Pull request merged**